### ISSUE TITLE
Ability to override Collection Auth

### ISSUE THREAD

**Issue by lil5** [usebruno/bruno#816]
1. I set the `Collection` `Auth` to `bearer token`
2. I click on a request and go to the `Auth` tab

### Expected result

I expect a list starting with `Collection default - Bearer Token` (selected), `AWS sigV4`, `Basic Auth`, `Bearer Token`, `No Auth`.

### Result

I get this list: `AWS sigV4`, `Basic Auth`, `Bearer Token`, `No Auth` (selected).

When `No Auth` is selected, the Collection default is used. I think this is confusing.

**Comment by btgs-0** [comment]
Hi. I have a WIP feature branch to implement a specific auth selection for this case: https://github.com/btgs-0/bruno/tree/feature/auth-from-collection

![image](https://github.com/usebruno/bruno/assets/149745752/f09bcee2-6763-4ef3-bd52-e5c38d745b77)
![image](https://github.com/usebruno/bruno/assets/149745752/1ce84553-0290-4314-859f-c527d2dd1cd5)

Let me know what you think

**Comment by lil5** [comment]
@btgs-0 The code looks good!

--- 

I'm unable to build your fork, I believe its due to the way the electron `packages/bruno-electron/package.json` doesn't install any `@usebruno/*` packages with `npm i` (I had a look at the `ls -ahl packages/bruno-electron/node_modules`).

I'm probably forgetting an install step.

Application log:
```log
Uncaught Exception:
Error: Cannot find module '/Users/lil5/Desktop/bruno/packages/bruno-electron/out/mac-arm64/Bruno.app/Contents/Resources/app.asar/node_modules/@usebruno/query/dist/cjs/index.js'. Please verify that the package.json has a valid "main" entry
at tryPackage (node:internal/modules/cjs/loader:360:19)
at Module._findPath (node:internal/modules/cjs/loader:573:18)
at Module._resolveFilename (node:internal/modules/cjs/loader:926:27)
at n._resolveFilename (node:electron/js2c/browser_init:249:1105)
at Module._load (node:internal/modules/cjs/loader:785:27)
at c._load (node:electron/js2c/asar_bundle:5:13339)
at Module.require (node:internal/modules/cjs/loader:1012:19)
at require (node:internal/modules/cjs/helpers:102:18)
at Object.<anonymous> (/Users/lil5/Desktop/bruno/packages/bruno-electron/out/mac-arm64/Bruno.app/Contents/Resources/app.asar/node_modules/@usebruno/js/src/utils.js:2:17)
at Module._compile (node:internal/modules/cjs/loader:1120:14)
```

**Comment by lil5** [comment]
@btgs-0 as this would be a breaking change I suggest adding a migration (`bruno.json` version: **1** -> **2** ) to change none to parent in bru files, this would keep the current functionality.

**Comment by btgs-0** [comment]
@lil5 that's a great idea, thanks. Your comment compelled me to double check export function and I've found that my branch does not export the collection auth to .bru correctly

**Comment by btgs-0** [comment]
I've opened a pull request: #907

**Comment by lil5** [comment]
To keep it on version 1 the auth schema could include a

```diff
const authSchema = Yup.object({
  mode: Yup.string().oneOf(['none', 'awsv4', 'basic', 'bearer', 'digest']).required('mode is required'),
+  collectionOverride: Yup.boolean(),
  awsv4: authAwsV4Schema.nullable(),
  basic: authBasicSchema.nullable(),
  bearer: authBearerSchema.nullable(),
  digest: authDigestSchema.nullable()
})
```

that way you don't need to include any difficult migration logic.

**Comment by btgs-0** [comment]
@lil5 will my branch cause a breaking change in the auth schema? since new auth type only applies to new requests, and I've only changed the schema to add the new auth type to it

``` diff
const authSchema = Yup.object({
+ mode: Yup.string().oneOf(['none', 'awsv4', 'basic', 'bearer', 'digest', 'parent']).required('mode is required'),
  awsv4: authAwsV4Schema.nullable(),
  basic: authBasicSchema.nullable(),
  bearer: authBearerSchema.nullable(),
  digest: authDigestSchema.nullable()
})
  .noUnknown(true)
  .strict();
```

**Comment by lil5** [comment]
It will break old Bruno projects that rely on auth type none to default to the collection auth. 

These hard decisions the project owner should make:
1. Add a migration function that reads the bruno.json file for which version it is and, on the users permission, migrate the collection to the latest format.
2. Add a key to the auth schema, it's more messy but it works. 
3. Handles both version 1 and 2 or 1.2 and only fix this issue on the next version.
4. Break it for the old version anyway. 
5. ny other suggestions?

**Comment by shille** [comment]
Postman carries Auth at the Folder level, can we please implement this where Parent has Auth and all levels can inherit from the parent be it the folder or the collection if it is the parent.  This should be supported for the Postman import as well.  Can raise a new feature request if required.

**Comment by lil5** [comment]
@shille 
Although adding folder level auth, is good to keep in mind! It requires a lot more change in the frontend unless Anoop is up for reviewing a much larger PR I suggest splitting your suggestion to a separate issue/pr

**Comment by btgs-0** [comment]
@shille it's a good idea, but like @lil5 said I think it would be better as a separate issue/pr. If my pr #907 gets pulled in it would cover 90+% of use cases

**Comment by sanjai0py** [comment]
This [PR](https://github.com/usebruno/bruno/pull/1667) implementing the feature has already been merged and released in [v1.10.0](https://github.com/usebruno/bruno/releases/tag/v1.10.0).

Now, there is an option in the request auth panel where the user can select `Inherit`, which inherits the auth used in the parent collection.

https://github.com/usebruno/bruno/assets/84461672/74c15d4b-3461-4d1a-b9c2-b0cbeb8edf4e
